core/src/test/java/org/elasticsearch/search/internal/InternalSearchHitTests.java

```diff
@@ -94,4 +213,41 @@ public void testHasSource() {
         searchHit.sourceRef(new BytesArray("{}"));
         assertTrue(searchHit.hasSource());
     }
+
+    public void testParseShardId() {
+        String indexName = randomAsciiOfLengthBetween(3,50);
+        int shardId = randomInt();
+        ShardId id = InternalSearchHit.parseShardId("["+indexName+"]["+shardId+"]", new XContentLocation(0, 0));
+        assertEquals(indexName, id.getIndexName());
+        assertEquals(shardId, id.getId());
+        assertEquals(indexName, id.getIndex().getName());
+        assertEquals(ClusterState.UNKNOWN_UUID, id.getIndex().getUUID());
+
+        ParsingException ex = expectThrows(ParsingException.class,
+                () -> InternalSearchHit.parseShardId(indexName + "," + shardId, new XContentLocation(0, 0)));
+        assertEquals("Failed to parse \"_shard\": expecting field with format [[indexName][shardId]] but found [" + indexName + ","
+                + shardId + "]", ex.getMessage());
+
+        ex = expectThrows(ParsingException.class,
+                () -> InternalSearchHit.parseShardId("["+indexName+"]["+shardId+"][somethingElse]", new XContentLocation(0, 0)));
+        assertEquals("Failed to parse \"_shard\": expecting field with format [[indexName][shardId]] but found [[" + indexName + "]["
+                + shardId + "][somethingElse]]", ex.getMessage());
+    }
+
+    private static Explanation createExplanation(int depth) {
+        String description = randomAsciiOfLengthBetween(5, 20);
+        float value = randomFloat();
+        List<Explanation> details = new ArrayList<>();
+        if (depth > 0) {
+            int numberOfDetails = randomIntBetween(1, 3);
+            for (int i = 0; i < numberOfDetails; i++) {
+                details.add(createExplanation(depth - 1));
+            }
+        }
+        if (randomBoolean()) {
+            return Explanation.match(value, description, details);
+        } else {
+            return Explanation.noMatch(description, details);
```

Comment:
I don't foresee us having that here. The search api can only give explanations about hits, that are implicitly a match. The explain api on the other hand explains also why a certain document doesn't match, in which case it does have a match boolean field. We are good without this.